src/main/java/net/dries007/tfc/common/blocks/berry_bush/BerryBush.java

```diff
@@ -0,0 +1,115 @@
+package net.dries007.tfc.common.blocks.berry_bush;
+
+import java.util.function.Supplier;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.Items;
+
+import net.dries007.tfc.common.items.TFCItems;
+
+import static net.dries007.tfc.common.blocks.berry_bush.AbstractBerryBushBlock.Lifecycle.*;
+
+public class BerryBush
```

Comment:
This needs some redesign
- Noop bush is confusing
- There's no json here, so no reason for a `T` / `T.Default` pattern
- Having both `Berry` and `BerryBush` is redundant
- `BerryBush.Default` is just a passthrough for `BerryBush` which is not an enum for some reason
- Nowhere do blocks actually use `getBush` meaning the type object doesn't need to be held on in the block, and can just be used to instantiate with req'd properties - i.e. see soil block variants. They don't hold on to their type objects.

Reply:
Berry is just a temp enum for me to register berry items. It will eventually be deleted when we start registering food.

The noop bit is essentially a fiction I created for inheritance purposes. Nearly everything fruit tree and berry bush related is a subclass of AbstractBerryBushBlock. So you need to be able to sometimes halt the previous logic